springfox-core/src/main/java/springfox/documentation/builders/OperationBuilder.java

```diff
@@ -232,9 +233,20 @@ public OperationBuilder tags(Set<String> tags) {
     return this;
   }
 
+  /**
+   * Sets the nickname for the operation
+   *
+   * @param nickname - a string for the nickname
+   * @return this
+   */
+  public OperationBuilder nickname(String nickname) {
+      this.nickname = defaultIfAbsent(nickname, this.nickname);
+      return this;
+  }
+
   public Operation build() {
     return new Operation(method, summary, notes, responseModel, uniqueId, position, tags, produces,
-        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden);
+        consumes, protocol, securityReferences, parameters, responseMessages, deprecated, isHidden, nickname);
```

Comment:
What I mean by that is that its fine for us to generate an operationId in swagger 2.0 regardless. It is optional after-all and library users can change the name if they wish, either via @who's solution or by registering in their own ```OperationIdGenerator``` which is an ```OperationBuilderPlugin```

Also, I'd like to be a little more deliberate about introducing configuration options in Docket. For all intents and purposes, if we're doing it right, its going to be the source for all breaking changes.

I'd like to keep the extensibility hooks very narrow and as few as possible.